macOS: Use `-hidden-l` to avoid re-exporting LLVM symbols.

maleadt opened the pull request:
This is equivalent to (the already used) `--exclude-libs,ALL` on Unix.

franz commented:
> I ran into linker errors with the old order, hence changing it.

Hmm ok, I wonder if the order should be different based on static/shared LLVM. Yes, linking LLVM is a mess..

> It is not unused, there's a target_link_directories using it

Yes, but except `target_link_directories` the only other command is the first `set` (that sets it to empty state). I don't see any command anywhere, that would put some actual content into the variable, or did i miss something ?

> Moving them to LLVM.cmake will require introducing separate _LIBS and _DIRS flags for both Clang's and LLVM's libraries though, so that will be a larger change

Yeah i think that's fine.

maleadt commented:
> I don't see any command anywhere, that would put some actual content into the variable, or did i miss something ?

Oh, the `list(APPEND POCL_PRIVATE_LINK_DIRECTORIES ${LIB_DIR})` somehow got lost.

I've added it back and reworked the change based on your suggestions.

franz reviewed (approved):
LGTM. I'm a bit wondering if we need the _DIRECTORIES variables when we have the LLVM_LIBDIR variable, but i'm not familiar with LLVM directory layout on Mac OS X so maybe it's necessary.
